packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -1471,6 +1471,11 @@ export interface BucketProps {
    * all objects in the bucket being deleted. Be sure to update your bucket resources
    * by deploying with CDK version `1.126.0` or later **before** switching this value to `false`.
    *
+   * Setting the `autoDeleteObjects: true` on a bucket will include `s3:PutBucketPolicy` action
+   * to the Bucket Policy. The reason being upon bucket deletion, the custom resource provider
+   * needs to update the bucket policy by adding a `DENY` policy for `s3:PutObject` to
+   * prevent a race condition on emptying with external bucket writers.
+   *
```

Comment:
```suggestion
   * Setting `autoDeleteObjects` to true on a bucket will add `s3:PutBucketPolicy` to the
   * bucket policy. This is because during bucket deletion, the custom resource provider
   * needs to update the bucket policy by adding a deny policy for `s3:PutObject` to
   * prevent race conditions with external bucket writers.
   *
```